android/java/org/chromium/chrome/browser/brave_news/BraveNewsUtils.java

```diff
@@ -38,17 +42,74 @@
 import java.util.Locale;
 import java.util.Map;
 
+@NullMarked
 public class BraveNewsUtils {
     public static final int BRAVE_NEWS_VIEWD_CARD_TIME = 1000; // milliseconds
-    private static Map<Integer, DisplayAd> sCurrentDisplayAds;
-    private static String mLocale;
-    private static List<Channel> mChannelList;
-    private static List<Publisher> sGlobalPublisherList;
-    private static List<Publisher> mPublisherList;
-    private static List<Channel> mFollowingChannelList;
-    private static List<Publisher> mFollowingPublisherList;
-    private static List<String> mSuggestionsList;
-    private static HashMap<String, Integer> mChannelIcons = new HashMap<>();
+    private static @Nullable Map<Integer, DisplayAd> sCurrentDisplayAds;
+    private static @Nullable String sLocale;
+    private static @Nullable List<Channel> sChannelList;
+    private static @Nullable List<Publisher> sGlobalPublisherList;
+    private static @Nullable List<Publisher> sPublisherList;
+    private static @Nullable List<Channel> sFollowingChannelList;
+    private static @Nullable List<Publisher> sFollowingPublisherList;
+    private static @Nullable List<String> sSuggestionsList;
+    private static HashMap<String, Integer> sChannelIcons = new HashMap<>();
+
+    // TODO(sergz): Ideally we want to move all news related files into it's
```

Comment:
I agree, but not just its own component, this class should not exist at all and defeats the purpose of what we're trying to do with these mojo interfaces. We absolutely should not be doing anything like this so can you also add to the comments that this entire class is deprecated and no one else should follow this pattern?

Reply:
also I'm going to create a follow-up issues for brave news because this class was a serious problem before multiple profiles were coming and now it's an even bigger problem

Reply:
https://github.com/brave/brave-browser/issues/45659